From f06e77eadd4134d92839168519f7ffdc56e7aa93 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sat, 03 Sep 2022 09:31:14 +0300
Subject: [PATCH] [SW-102303] gaudi3: modify fuse read return value

Currently, fuse init return value is explicitly check to 0 else
ROM enters in halt state. if fuse init is already done (checked
by reading cold flops) fuse_init function returns -1 which will
cause rom to enter halt state. This patch change return value
to 0 in case fuse init is already done.

Change-Id: I63293845ff0762a9d348f592d0398e90d8d6a681
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 9327cac..c10ea9c 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -126,8 +126,11 @@
 {
 	struct gaudi3_cold_rst_data cold_rst_data;
 
-	/* efuse init successful, set cold flops */
 	cold_rst_data.data = regmap_readl(mmCOLD_RST_FLOPS_2);
+	if (cold_rst_data.efuse_read_flag)
+		return;
+
+	/* efuse init successful, set cold flops */
 	cold_rst_data.efuse_read_flag = 1;
 	regmap_writel(cold_rst_data.data, mmCOLD_RST_FLOPS_2);
 }
diff --git a/common/efuse.c b/common/efuse.c
index e470018..e7d699d 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -42,7 +42,7 @@
 	fuse_prog_range_config();
 
 	if (arcboot_rom_global.efuse_read == 1)
-		return -1;
+		return 0;
 
 	read_reg = mmEFUSE_READ_CTRL;
 
